output/file/outputfile.go

```diff
@@ -0,0 +1,219 @@
+	OpenFile(name string, flag int, perm os.FileMode) (file, error)
+}
+
+type file interface {
+	io.Closer
+	io.Reader
+	io.ReaderAt
+	io.Seeker
+	Sync() error
+	Stat() (os.FileInfo, error)
+	Write(b []byte) (n int, err error)
+}
+
+// osFS implements fileSystem using the local disk. this is the default implementation to use, except during when we mock
+type osFS struct{}
+
+func (osFS) Open(name string) (*os.File, error) { return os.Open(name) }
+func (osFS) Stat(name string) (fileInfo, error) { return os.Stat(name) }
+func (osFS) MkdirAll(path string, perm os.FileMode) error {
+	return os.MkdirAll(path, perm)
+}
+func (osFS) OpenFile(name string, flag int, perm os.FileMode) (file, error) {
+	return os.OpenFile(name, flag, perm)
+}
+
+// OutputConfig holds the configuration json fields and internal objects
+type OutputConfig struct {
+	config.OutputConfig
+	CreateIfDeleted bool   `json:"create_if_deleted"` // If the configured file is deleted, but an event is handled by the plugin, the plugin will recreate the file. Default ⇒ true
+	DirMode         string `json:"dir_mode"`          // Dir access mode to use. Example: "dir_mode" => 0750
+	FileMode        string `json:"file_mode"`         // File access mode to use. Example: "file_mode" => 0640
+	FlushInterval   int    `json:"flush_interval"`    // Flush interval (in seconds) for flushing writes to log files. 0 will flush on every message.
+	Path            string `json:"path"`              // The path to the file to write. Event fields can be used here, like /var/log/logstash/%{host}/%{application}
+	Codec           string `json:"codec"`             // expression to write to file. E.g. "%{log}"
+	WriteBehavior   string `json:"write_behavior"`    // If append, the file will be opened for appending and each new event will be written at the end of the file. If overwrite, the file will be truncated before writing and only the most recent event will appear in the file.
+	writers         map[string]chan string
+	fileMode        os.FileMode
+	dirMode         os.FileMode
+	fs              fileSystem
+}
+
+// DefaultOutputConfig returns an OutputConfig struct with default values
+func DefaultOutputConfig() OutputConfig {
+	return OutputConfig{
+		OutputConfig: config.OutputConfig{
+			CommonConfig: config.CommonConfig{
+				Type: ModuleName,
+			},
+		},
+		CreateIfDeleted: defaultCreateIfDeleted,
+		DirMode:         defaultDirMode,
+		FileMode:        defaultFileMode,
+		FlushInterval:   defaultFlushInterval,
+		WriteBehavior:   defaultWriteBehavior,
+		Codec:           defaultCodec,
+		writers:         make(map[string]chan string),
+		fs:              osFS{},
+	}
+}
+
+// InitHandler initialize the output plugin
+func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeOutputConfig, error) {
+	conf := DefaultOutputConfig()
+	err := config.ReflectConfig(raw, &conf)
+	if err != nil {
+		return nil, err
+	}
+	if conf.Path == "" {
+		return nil, errors.New("no path defined for file")
```

Comment:
It's hard to catch this kind of error. You can reference [output/redis/outputredis.go](https://github.com/tsaikd/gogstash/blob/master/output/redis/outputredis.go) for error handling.

Reply:
thanks; updated.